packages/runtime/src/types.ts

```
@@ -92,8 +92,69 @@ export type GatewayPlugin<
 > = YogaPlugin<Partial<TPluginContext> & GatewayContext & TContext> &
   UnifiedGraphPlugin<Partial<TPluginContext> & GatewayContext & TContext> & {
     onFetch?: OnFetchHook<Partial<TPluginContext> & GatewayContext & TContext>;
+    onCacheGet?: OnCacheGetHook;
+    onCacheSet?: OnCacheSetHook;
+    onCacheDelete?: OnCacheDeleteHook;
   } & Partial<Disposable | AsyncDisposable>;
 
+export type OnCacheGetHook = (
+  payload: OnCacheGetHookEventPayload,
+) => MaybePromise<OnCacheGetHookResult | void>;
```

Biome (1.9.4)

[error] void is confusing inside a union type.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)